modules/docs/arrow-docs/docs/docs/arrow/typeclasses/traverse/README.md

```diff
@@ -11,16 +11,495 @@ redirect_from:
+import arrow.typeclasses.Foldable
+import arrow.typeclasses.Functor
+
+interface Traverse<F> : Functor<F>, Foldable<F> {
+  fun <G, A, B> Kind<F, A>.traverse(AP: Applicative<G>, f: (A) -> Kind<G, B>): Kind<G, Kind<F, B>>
+}
+```
+
+In our above example, `F` is `List`, and `G` is `Option`, `Either`, or `Promise`. For the profile example, traverse says given a `List<User>` and a function `(User) -> Promise<ForIO, Profile>`, it can give you a `Promise<ForIO, List<Profile>>`.
+
+Abstracting away the `G` (still imagining `F` to be `List`), `traverse` says given a collection of data, and a function that takes a piece of data and returns an value `B` wrapped in a container `G`, it will traverse the collection, applying the function and aggregating the values (in a `List`) as it goes.
+
+In the most general form, `Kind<F, A>` is some sort of context `F` which may contain a value `A` (or several). While `List` tends to be among the most general cases, there also exist `Traverse` instances for `Option`, `Either`, and `Validated` (among others).
+
+Essentially, one uses `Traverse`, when there is a function `(A) -> Kind<G, B>` and `G` is an `Applicative` instance, thus `G` provides `just` and `map` functions e.g.: just as `Promise<ForIO, B>` provides and you want to `lift` the previous function to work on `Kind<F, A>` where `F` is a `Traverse` instance - and you still care about `A` (the `Profiles`) `(List<User>)` and return `Kind<G, Kind<F, B>>` e.g.: `Promise<ForIO, List<Profile>>`
+
+### To fold Or not to fold
+
+Even though, `Foldable` and `Traverse` are related, because both 'reduce their values to something', it is not obvious why to consider `Traverse` over `Foldable`.
+
+Here is a small example:
+
+```kotlin:ank:playground
+import arrow.core.MapK
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.fix
+import arrow.core.k
+
+fun main(){
+  //sampleStart
+  val map: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+
+  val optionMapK: Option<MapK<String, String>> = map.traverse(Option.applicative()) { Some("$it") }.fix()
+  
+  // val optionMapKBoilered = map.foldLeft(Some(emptyMap())) { acc: Option<MapK<String, String>>, i: Int ->
+  //   acc.fold({ emptyMap() }, { /*Some logic to retrieve the key of a value, transform it and add it to the accumulated Map*/ })
+  // }
+  //sampleEnd
+  println(optionMapK)
+}
+```
+
+Both values try to attain the same thing, but what `Foldable` lacks is that it solely drills down to it's `A` here `Int` and does not preserve its shape `F` - `MapK`. This is where `Traverse` shines, whenever you care about the Output `B` from `(A) -> B` and the existing shape of `F` you would use `traverse`.
+
+Additionally, you're able to wrap your context `F` within a `G`. That is, may among others, the reason why `Traverse` is strictly more powerful than `Foldable`.
+
+You can also misuse it's powers - where a `map` is more considerable: 
```

Comment:
What do you mean with this?

Reply:
Here I wanted to give a counter-example, how not to use `Traverse`. Here it results in an inefficient implementation because we `wrap` and `unwrap` the `Int`

Reply:
I think I had to add that exact sentence from above to the markdown, haha